doc/decisions/notifications.md

```diff
@@ -0,0 +1,39 @@
+# Notifications
+
+## Problem
+
+Ensure (contracts) shifted expectations:
+Applications want control over if they receive notifications.
+
+## Constraints
+
+- applications need to send notifications in any case, otherwise others cannot receive it
+
+## Assumptions
+
+- multiple transport mechanism at once do not need to be supported
+
+## Considered Alternatives
+
+## Decision
+
+Have a symlink to which transport notification plugin is used.
+This notification plugin is used by default to sent notifications.
+
+If applications also want to receive notifications is up to the contract.
+
+## Rationale
+
+## Implications
+
+The decision which transport mechanism, e.g. dbus or zeromq, must be
+chosen per computer node or container. Within one Elektra setup,
+they cannot be intermixed (unlike config file formats, where many
```

Comment:
> Within one Elektra setup, they cannot be intermixed

Is there a technical reason, or would sending notifications via multiple transports at the same time (e.g. both `dbus` and `zeromq`) work, i.e. all messages duplicated to all transports?

If multiple transports is technically possible, we shouldn't use a `symlink` and instead add config keys in `system:/elektra`.

Reply:
It simply follows the simple rule to keep everything simple as long as nobody needs something more complicated. If someone *really* wants to use several transports on one system, the person can also use some hub that reads from one transport and relays to another. So we do not break any use case by reducing our notification system to one transport.

Reply:
I agree the symlink makes sense to indicate the default (like with resolver and storage). But the decision (at least to me) sounds like we are saying we will never support multiple transports. I still don't see a technical reason why we should restrict ourselves to a single notification transport forever.

I would suggest, we decide that
- the symlink indicates the default notification transport
- we currently only support one transport at a time, so the default is always used
- in future, if the need arises, there may be additional options (e.g. config keys) to enable secondary transports

Reply:
Yes, of course! The decisions always only describe what we have now or want to implement for 1.0. Decisions can change, especially if assumptions are wrong. That is why it is important to write down assumptions.

So I think this decision describes what we both want.

Reply:
> The decisions always only describe what we have now or want to implement for 1.0. Decisions can change, especially if assumptions are wrong.

But we have to be careful, how we write things. Future developers (maybe we ourselves), might just read the decision and use it as the basis for their own decision. "must be chosen per computer node or container" sounds very much final and could result in APIs or implementation details (in the notification stuff, or elsewhere)  that may be very hard to change without breaking things.

Reply:
Yes, of course assumptions need to be chosen carefully. They might lead to unsuitability of Elektra in some areas. But as you often stated: we also cannot please everybody, so some assumptions are needed to make Elektra usable.